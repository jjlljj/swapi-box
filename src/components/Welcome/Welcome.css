.welcome-crawl {
  font-family: 'Pathway Gothic One', sans-serif;
  background-size: cover;
  color: #fedb4d;
  display: flex;
  font-size: 72px;
  font-weight: bold;
  height: 80vh;
  justify-content: center;
  letter-spacing: 8px;
  line-height: 140%;
  min-height: 400px;
  overflow: hidden;
  perspective: 400px;
  position: relative;
  text-align: justify;
  width: 100%;
}

.crawl-wrap {
  position: relative;
  transform-origin: 50% 100%;
  animation: crawl 70s linear infinite;
}

.title-wrap {
  text-align: center;
}

.title-wrap h2 {
  font-size: 100px;
  font-weight: 800;
  margin: 40px;
  text-transform: uppercase;
}

@keyframes crawl {
  0% {
    top: 600px;
    transform: rotateX(20deg) translateZ(0);
  }

  100% {
    top: -5000px;
    transform: rotateX(25deg) translateZ(-2500px);
  }
}


@keyframes move-animate-back {
    from {background-position:0 0;}
    to {background-position:-10000px 5000px;}
}

.stars, 
.stars-animate {
  height:100%;
  left:0;
  position:absolute;
  top:0;
  width:100%;
}

.stars {
  background:#000 url(./assets/stars.png) repeat top center;
  z-index:-1;
}

.stars-animate{
  background:transparent url(./assets/animate.png) repeat top center;
  z-index:0;

  animation:move-animate-back 300s linear infinite;
}
