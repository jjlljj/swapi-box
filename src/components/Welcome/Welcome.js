import React from 'react';
import { string } from 'prop-types';
import './Welcome.css';

const Welcome = ({ episode, title, text}) => {

  return (
    <section className="welcome-crawl"> 
      <div className="stars"></div>
      <div className="stars-animate"></div>
      <div className="fade"></div>
      <div className="crawl-wrap">
        <div className="title-wrap">
          <h4>{episode}</h4>
          <h2>{title}</h2>
        </div>
        <p>{text}</p>
      </div>
    </section>
  );
};

Welcome.propTypes = {
  episode: string,
  title: string,
  text: string
};

export default Welcome;
