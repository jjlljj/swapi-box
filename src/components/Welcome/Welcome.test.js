/* eslint-disable */

import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import Welcome from './Welcome';

describe('Welcome', () => {

  it('should match snapshot', () => {
    const mockMovie = {
      episode: 'Episode IV',
      title: 'A New Hope',
      text: 'It is a period of civil war...'
    }
    const renderendComponent = shallow(<Welcome {...mockMovie} />)

    expect(renderendComponent).toMatchSnapshot()
  })

  it('should match snapshot when passed different props', () => {
    const mockMovie = {
      episode: 'Episode V',
      title: 'The Empire Strikes Back',
      text: 'After the ...'
    }
    const renderendComponent = shallow(<Welcome {...mockMovie} />)

    expect(renderendComponent).toMatchSnapshot()
  })

  it('should match snapshot when not passed props', () => {
    const renderendComponent = shallow(<Welcome />)

    expect(renderendComponent).toMatchSnapshot()
  })

})
