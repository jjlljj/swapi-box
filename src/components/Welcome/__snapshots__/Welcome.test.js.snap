// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    episode="Episode IV"
    text="It is a period of civil war..."
    title="A New Hope"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stars"
/>,
        <div
          className="stars-animate"
/>,
        <div
          className="fade"
/>,
        <div
          className="crawl-wrap"
>
          <div
                    className="title-wrap"
          >
                    <h4>
                              Episode IV
                    </h4>
                    <h2>
                              A New Hope
                    </h2>
          </div>
          <p>
                    It is a period of civil war...
          </p>
</div>,
      ],
      "className": "welcome-crawl",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars-animate",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title-wrap"
>
              <h4>
                            Episode IV
              </h4>
              <h2>
                            A New Hope
              </h2>
</div>,
            <p>
              It is a period of civil war...
</p>,
          ],
          "className": "crawl-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Episode IV
</h4>,
                <h2>
                  A New Hope
</h2>,
              ],
              "className": "title-wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Episode IV",
                },
                "ref": null,
                "rendered": "Episode IV",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "A New Hope",
                },
                "ref": null,
                "rendered": "A New Hope",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "It is a period of civil war...",
            },
            "ref": null,
            "rendered": "It is a period of civil war...",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stars"
/>,
          <div
            className="stars-animate"
/>,
          <div
            className="fade"
/>,
          <div
            className="crawl-wrap"
>
            <div
                        className="title-wrap"
            >
                        <h4>
                                    Episode IV
                        </h4>
                        <h2>
                                    A New Hope
                        </h2>
            </div>
            <p>
                        It is a period of civil war...
            </p>
</div>,
        ],
        "className": "welcome-crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars-animate",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title-wrap"
>
                <h4>
                                Episode IV
                </h4>
                <h2>
                                A New Hope
                </h2>
</div>,
              <p>
                It is a period of civil war...
</p>,
            ],
            "className": "crawl-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Episode IV
</h4>,
                  <h2>
                    A New Hope
</h2>,
                ],
                "className": "title-wrap",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Episode IV",
                  },
                  "ref": null,
                  "rendered": "Episode IV",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "A New Hope",
                  },
                  "ref": null,
                  "rendered": "A New Hope",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "It is a period of civil war...",
              },
              "ref": null,
              "rendered": "It is a period of civil war...",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Welcome should match snapshot when not passed props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stars"
/>,
        <div
          className="stars-animate"
/>,
        <div
          className="fade"
/>,
        <div
          className="crawl-wrap"
>
          <div
                    className="title-wrap"
          >
                    <h4 />
                    <h2 />
          </div>
          <p />
</div>,
      ],
      "className": "welcome-crawl",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars-animate",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title-wrap"
>
              <h4 />
              <h2 />
</div>,
            <p />,
          ],
          "className": "crawl-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4 />,
                <h2 />,
              ],
              "className": "title-wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h2",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stars"
/>,
          <div
            className="stars-animate"
/>,
          <div
            className="fade"
/>,
          <div
            className="crawl-wrap"
>
            <div
                        className="title-wrap"
            >
                        <h4 />
                        <h2 />
            </div>
            <p />
</div>,
        ],
        "className": "welcome-crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars-animate",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title-wrap"
>
                <h4 />
                <h2 />
</div>,
              <p />,
            ],
            "className": "crawl-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4 />,
                  <h2 />,
                ],
                "className": "title-wrap",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": undefined,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Welcome should match snapshot when passed different props 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Welcome
    episode="Episode V"
    text="After the ..."
    title="The Empire Strikes Back"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="stars"
/>,
        <div
          className="stars-animate"
/>,
        <div
          className="fade"
/>,
        <div
          className="crawl-wrap"
>
          <div
                    className="title-wrap"
          >
                    <h4>
                              Episode V
                    </h4>
                    <h2>
                              The Empire Strikes Back
                    </h2>
          </div>
          <p>
                    After the ...
          </p>
</div>,
      ],
      "className": "welcome-crawl",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "stars-animate",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fade",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="title-wrap"
>
              <h4>
                            Episode V
              </h4>
              <h2>
                            The Empire Strikes Back
              </h2>
</div>,
            <p>
              After the ...
</p>,
          ],
          "className": "crawl-wrap",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Episode V
</h4>,
                <h2>
                  The Empire Strikes Back
</h2>,
              ],
              "className": "title-wrap",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Episode V",
                },
                "ref": null,
                "rendered": "Episode V",
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "The Empire Strikes Back",
                },
                "ref": null,
                "rendered": "The Empire Strikes Back",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "After the ...",
            },
            "ref": null,
            "rendered": "After the ...",
            "type": "p",
          },
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="stars"
/>,
          <div
            className="stars-animate"
/>,
          <div
            className="fade"
/>,
          <div
            className="crawl-wrap"
>
            <div
                        className="title-wrap"
            >
                        <h4>
                                    Episode V
                        </h4>
                        <h2>
                                    The Empire Strikes Back
                        </h2>
            </div>
            <p>
                        After the ...
            </p>
</div>,
        ],
        "className": "welcome-crawl",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "stars-animate",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fade",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title-wrap"
>
                <h4>
                                Episode V
                </h4>
                <h2>
                                The Empire Strikes Back
                </h2>
</div>,
              <p>
                After the ...
</p>,
            ],
            "className": "crawl-wrap",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h4>
                    Episode V
</h4>,
                  <h2>
                    The Empire Strikes Back
</h2>,
                ],
                "className": "title-wrap",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Episode V",
                  },
                  "ref": null,
                  "rendered": "Episode V",
                  "type": "h4",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "The Empire Strikes Back",
                  },
                  "ref": null,
                  "rendered": "The Empire Strikes Back",
                  "type": "h2",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "After the ...",
              },
              "ref": null,
              "rendered": "After the ...",
              "type": "p",
            },
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
